scalyr_agent/builtin_monitors/syslog_utils.py

```diff
@@ -0,0 +1,211 @@
+
+        @param conf: Logger configuration in the form of a dictionary or JsonObject, that a matcher should be created for.
+        @return: Logger configuration in the form of a dictionary or JsonObject if this matcher matches the passed
+        in data, None otherwise
+        """
+        config = copy.deepcopy(conf)
+        if "syslog_app" not in config:
+            config["syslog_app"] = ".*"
+        if "parser" not in config:
+            config["parser"] = self._extra_config.get("parser")
+        if "attributes" not in config:
+            config["attributes"] = JsonObject({"monitor": six.text_type("agentSyslog")})
+        elif "monitor" not in config["attributes"]:
+            config["attributes"]["monitor"] = six.text_type("agentSyslog")
+        config["attributes"]["containerName"] = self._extra_config.get("containerName")
+        config["attributes"]["containerId"] = self._extra_config.get("containerId")
+        file_template = Template(self._extra_config.get("message_log"))
+        regex = re.compile(config["syslog_app"])
+        match_hash = six.text_type(hash(config["syslog_app"]))
+        if config["syslog_app"] == ".*":
+            match_hash = ""
+        full_path = os.path.join(
+            self._global_config.agent_log_path,
+            file_template.safe_substitute({"HASH": match_hash}),
+        )
+        matched_config = JsonObject({"parser": "syslog", "path": full_path})
+        matched_config.update(config)
+
+        def config_matcher(unit):
+            if regex.match(unit) is not None:
+                return matched_config
+            return None
+
+        return config_matcher
+
+    def get_logger(self, unit):
+        result = LogConfigManager.get_logger(self, unit)
+        self.log_deleter.add_current_log(self.get_config(unit)["path"])
+        return result
+
+    def check_for_old_logs(self):
+        self.log_deleter.check_for_old_logs()
+
+
+class LogDeleter(object):
+    """Deletes unused log files that match a log_file_template"""
+
+    def __init__(
+        self,
+        check_interval_mins,
+        delete_interval_hours,
+        check_rotated_timestamps,
+        max_log_rotations,
+        log_file_template,
+        agent_log_path,
+    ):
+        self._check_interval = check_interval_mins * 60
+        self._delete_interval = delete_interval_hours * 60 * 60
+        self._check_rotated_timestamps = check_rotated_timestamps
+        self._max_log_rotations = max_log_rotations
+        self._log_glob = os.path.join(
+            agent_log_path, log_file_template.safe_substitute(HASH="*")
+        )
+
+        self._last_check = time.time()
+        self.existing_logs = []
+
+    def add_current_log(self, path):
+        self.existing_logs.append(path)
+
+    def _get_old_logs_for_glob(
+        self, current_time, glob_pattern, existing_logs, check_rotated, max_rotations
```

Comment:
Docstrings + type annotations for all those arguments would be nice.

Reply:
EDIT: Now I see you moved this code from a different module - in any case, now is our opportunity to improve it :)

Reply:
Type annotations aren't supported for python 2 right? Or did you just mean document them in the docstring?

Reply:
They are supported using comments notation.